test/customization_test.rb

```diff
@@ -0,0 +1,292 @@
+      function test( int index, char character, float precision, double trouble )
+    LANG
+  end
+
+  it 'must work in a group if the line doesn\'t fit' do
+    printer = Oppen::Wadler.new(margin: 45)
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+        test(
+          int index, char character,
+          float precision, double trouble
+        )
+    LANG
+  end
+end
+
+describe 'Spaces tests' do
+  it 'must work with a string of length 1' do
+    printer = Oppen::Wadler.new(margin: 45, space: '*')
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+      **test(
+      ****int index, char character,
+      ****float precision, double trouble
+      **)
+    LANG
+  end
+
+  it 'must work with a string of length greater than 1' do
+    printer = Oppen::Wadler.new(margin: 45, space: '**')
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+      ****test(
+      ********int index, char character,
+      ********float precision, double trouble
+      ****)
+    LANG
+  end
+
+  it 'must work with a lambda' do
+    printer = Oppen::Wadler.new(margin: 45, space: ->(n) { '*' * n })
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+      **test(
+      ****int index, char character,
+      ****float precision, double trouble
+      **)
+    LANG
+  end
+
+  it 'must work with a lambda different from default' do
+    printer = Oppen::Wadler.new(margin: 45, space: ->(n) { '*' * n * 2 })
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+      ****test(
+      ********int index, char character,
+      ********float precision, double trouble
+      ****)
+    LANG
+  end
+
+  it 'must work with a proc' do
+    printer = Oppen::Wadler.new(margin: 45, space: proc { |n| '*' * n })
+    customization_build_output(printer)
+    _(printer.output).must_equal <<~LANG.chomp
+      function
+      **test(
+      ****int index, char character,
+      ****float precision, double trouble
+      **)
+    LANG
+  end
+
+  it 'must raise an error for a lambda of arity different than one' do
+    printer = Oppen::Wadler.new(margin: 45, space: ->(n, _o) { '*' * n })
+    _ { printer.output }.must_raise ArgumentError
+
+    printer = Oppen::Wadler.new(margin: 45, space: -> { '*' })
+    _ { printer.output }.must_raise ArgumentError
+
+    printer = Oppen::Wadler.new(margin: 45, space: ->(*_args) { '*' })
+    _ { printer.output }.must_raise ArgumentError
+  end
+end
+
+describe 'Line delimiter tests' do
+  it 'must work with a \n delimiter' do
+    printer = Oppen::Wadler.new(margin: 45, new_line: '\n')
+    customization_build_output(printer)
+    _(printer.output)
+      .must_equal 'function\n  test(\n    int index, char character,\n    float precision, double trouble\n  )'
+  end
```

Comment:
Nah, don't use `\n`. It's confusing. Use a visible ASCII char.

Actually, now that I think about it, we should also add tests on UTF-8 chars outside the ASCII range.